pkg/reckoner/plot.go

```diff
@@ -157,6 +150,27 @@ func (c Client) TemplateRelease(releaseName string) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("error templating release %s: %s", releaseName, stdErr)
 	}
+
+	if len(templateOutputDir) > 0 {
+		err = c.WriteSplitYaml([]byte(out), templateOutputDir, releaseName)
+		if err != nil {
+			color.Red(err.Error())
+			os.Exit(1)
+		}
+	}
+
+	// if the user has specified a RepoURL, they probably also
+	// specified other things. Most required things receive a default
+	// and a warning when not found in the config file. This
+	// maybe could use some improvement
+	if c.CourseFile.GitOps.ArgoCD.Spec.Source.RepoURL != "" && templateOutputDir != "" {
```

Comment:
Would it be better to do 

```
if _, ok := c.CourseFile.GitOps.ArgoCD ; ok {

  // Write out the argocd Applications
```

This could go hand-in hand with adding the outputDir to the client struct, and then check if it is set inside of the writeArgoApplications function.

Reply:
I'm not sure that works 🤔 please elaborate.

another idea I'll throw into the discussion: we could compare `c.CourseFile.GitOps.ArgoCD` to `course.ArgoApplication{}` and see if they (don't) match. if they don't match, something was modified via the course file. if they match, nothing has been given.

🤔 we could probably come up with some other ideas, too.

Reply:
Your idea seems good to me, and achieves what I was trying to suggest, or as we talked about today probably good to just leave as-is.